listenbrainz_spark/recommendations/tests/test_models.py

```diff
@@ -22,44 +25,158 @@ def tearDownClass(cls):
         super().delete_dir()
         super().tearDownClass()
 
-    def test_train(self):
-        training_data, validation_data, test_data = super().split_playcounts()
-
-        best_model, model_metadata, best_model_metadata = train_models.train(
-            training_data, validation_data, validation_data.count(), self.ranks,
-            self.lambdas, self.iterations, self.alpha
-        )
-        self.assertTrue(best_model)
-        self.assertEqual(len(model_metadata), 1)
-        self.assertEqual(model_metadata[0][0], best_model_metadata['model_id'])
-        self.assertEqual(model_metadata[0][1], best_model_metadata['training_time'])
-        self.assertEqual(model_metadata[0][2], best_model_metadata['rank'])
-        self.assertEqual(model_metadata[0][3], best_model_metadata['lmbda'])
-        self.assertEqual(model_metadata[0][4], best_model_metadata['iteration'])
-        self.assertEqual(model_metadata[0][5], best_model_metadata['error'])
-        self.assertEqual(model_metadata[0][6], best_model_metadata['rmse_time'])
-
     def test_parse_dataset(self):
-        row = Row(user_id=1, recording_id=1, count=1)
+        row = Row(user_id=1, recording_id=2, count=3)
         rating_object = train_models.parse_dataset(row)
         self.assertEqual(rating_object.user, 1)
-        self.assertEqual(rating_object.product, 1)
-        self.assertEqual(rating_object.rating, 1)
+        self.assertEqual(rating_object.product, 2)
+        self.assertEqual(rating_object.rating, 3)
+
+    @patch('listenbrainz_spark.recommendations.train_models.sqrt')
+    @patch('listenbrainz_spark.recommendations.train_models.RDD')
+    @patch('listenbrainz_spark.recommendations.train_models.train')
+    @patch('listenbrainz_spark.recommendations.train_models.add')
+    def test_compute_rmse(self, mock_add, mock_train, mock_rdd, mock_sqrt):
+        n = 1
+        model_id = "281c4177-f33a-441d-b15d-910acaf18b07"
+        _ = train_models.compute_rmse(mock_train.return_value, mock_rdd, n, model_id)
```

Comment:
I dont understand this -- you are using the return_value of the mock, but the mock hasn't been used yet...

Reply:
an actual model should be passed to `compute_rmse`.
the model is the return value of `train_model` function and not some kind of import that I can directly patch.

is there any other way to do that?

Reply:
Not really sure. I think having @paramsingh take a look would be good.

Reply:
Passing a `MagicMock` object would be clearer.

```
mock_model = MagicMock()
_ = train_models.compute_rmse(mock_model, mock_rdd, n, model_id)
```

Reply:
Thanks :)